#### Prerequisite

In [2]:
#Load SQL module

%load_ext sql

In [3]:
#Connect to the database

%sql mysql://root:Nemai1990@localhost:3306/mydatadb

'Connected: root@mydatadb'

## Data Exploration

Select data and take a look of the dataset

In [10]:
%%sql

SELECT *
FROM covid_deaths
LIMIT 1000;

 * mysql://root:***@localhost:3306/mydatadb
1000 rows affected.


[('AFG', 'Asia', 'Afghanistan', datetime.datetime(2020, 2, 24, 0, 0), 1.0, 1.0, None, None, None, None, 0.026, 0.026, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 8.33, 38928341.0, 54.422, 18.6, 2.581, 1.337, 1803.987, None, 597.029, 9.59, None, None, 37.746, 0.5, 64.83, 0.511),
 ('AFG', 'Asia', 'Afghanistan', datetime.datetime(2020, 2, 25, 0, 0), 1.0, 0.0, None, None, None, None, 0.026, 0.0, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 8.33, 38928341.0, 54.422, 18.6, 2.581, 1.337, 1803.987, None, 597.029, 9.59, None, None, 37.746, 0.5, 64.83, 0.511),
 ('AFG', 'Asia', 'Afghanistan', datetime.datetime(2020, 2, 26, 0, 0), 1.0, 0.0, None, None, None, None, 0.026, 0.0, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 8.33, 38928341.0, 54.422, 18.6, 2.581, 1.337, 1803.987, None, 597.029, 9.59, None, None, 37.746, 0.5, 64.83, 0.511),
 ('AFG', 'Asia', 'Afghanistan', datetime.datetime(2020, 2, 27, 0, 0), 1.0, 0.0, None, None, None, None, 0.026, 0.0, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 8.33, 38928341.0, 54.422, 18.6, 2.581, 1.337, 1803.987, None, 597.029, 9.59, None, None, 37.746, 0.5, 64.83, 0.511),
 ('AFG', 'Asia', 'Afghanistan', datetime.datetime(2020, 2, 28, 0, 0), 1.0, 0.0, None, None, None, None, 0.026, 0.0, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 8.33, 38928341.0, 54.422, 18.6, 2.581, 1.337, 1803.987, None, 597.029, 9.59, None, None, 37.746, 0.5, 64.83, 0.511),
 ('AFG', 'Asia', 'Afghanistan', datetime.datetime(2020, 2, 29, 0, 0), 1.0, 0.0, 0.143, None, None, 0.0, 0.026, 0.0, 0.004, None, None, 0.0, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 8.33, 38928341.0, 54.422, 18.6, 2.581, 1.337, 1803.987, None, 597.029, 9.59, None, None, 37.746, 0.5, 64.83, 0.511),
 ('AFG', 'Asia', 'Afghanistan', datetime.datetime(2020, 3, 1, 0, 0), 1.0, 0.0, 0.143, None, None, 0.0, 0.026, 0.0, 0.004, None, None, 0.0, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 27.78, 38928341.0, 54.422, 18.6, 2.581, 1.337, 1803.987, None, 597.029, 9.59, None, None, 37.746, 0.5, 64.83, 0.511),
 ('AFG', 'Asia', 'Afghanistan', datetime.datetime(2020, 3, 2, 0, 0), 1.0, 0.0, 0.0, None, None, 0.0, 0.026, 0.0, 0.0, None, None, 0.0, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 27.78, 38928341.0, 54.422, 18.6, 2.581, 1.337, 1803.987, None, 597.029, 9.59, None, None, 37.746, 0.5, 64.83, 0.511),
 ('AFG', 'Asia', 'Afghanistan', datetime.datetime(2020, 3, 3, 0, 0), 2.0, 1.0, 0.143, None, None, 0.0, 0.051, 0.026, 0.004, None, None, 0.0, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 27.78, 38928341.0, 54.422, 18.6, 2.581, 1.337, 1803.987, None, 597.029, 9.59, None, None, 37.746, 0.5, 64.83, 0.511),
 ('AFG', 'Asia', 'Afghanistan', datetime.datetime(2020, 3, 4, 0, 0), 4.0, 2.0, 0.429, None, None, 0.0, 0.103, 0.051, 0.011, None, None, 0.0, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None

As we can see, We have tons of data and many columns. We are not gonna work on all of them but few. Lets select toal_cases, new_cases, total_deaths and poplution. 

In [16]:
%%sql

SELECT location, date, total_cases, new_cases, total_deaths, population
FROM covid_deaths
ORDER BY 2
LIMIT 100;

 * mysql://root:***@localhost:3306/mydatadb
100 rows affected.


[('Mexico', datetime.datetime(2020, 1, 1, 0, 0), None, None, None, 128932753.0),
 ('Argentina', datetime.datetime(2020, 1, 1, 0, 0), None, None, None, 45195777.0),
 ('Argentina', datetime.datetime(2020, 1, 2, 0, 0), None, None, None, 45195777.0),
 ('Mexico', datetime.datetime(2020, 1, 2, 0, 0), None, None, None, 128932753.0),
 ('Mexico', datetime.datetime(2020, 1, 3, 0, 0), None, None, None, 128932753.0),
 ('Argentina', datetime.datetime(2020, 1, 3, 0, 0), None, None, None, 45195777.0),
 ('Argentina', datetime.datetime(2020, 1, 4, 0, 0), None, None, None, 45195777.0),
 ('Mexico', datetime.datetime(2020, 1, 4, 0, 0), None, None, None, 128932753.0),
 ('Thailand', datetime.datetime(2020, 1, 4, 0, 0), None, None, None, 69799978.0),
 ('Thailand', datetime.datetime(2020, 1, 5, 0, 0), None, None, None, 69799978.0),
 ('Argentina', datetime.datetime(2020, 1, 5, 0, 0), None, None, None, 45195777.0),
 ('Mexico', datetime.datetime(2020, 1, 5, 0, 0), None, None, None, 128932753.0),
 ('Mexico', datetime.datetime(2020, 1, 6, 0, 0), None, None, None, 128932753.0),
 ('Thailand', datetime.datetime(2020, 1, 6, 0, 0), None, None, None, 69799978.0),
 ('Peru', datetime.datetime(2020, 1, 6, 0, 0), None, None, None, 32971846.0),
 ('Argentina', datetime.datetime(2020, 1, 6, 0, 0), None, None, None, 45195777.0),
 ('Argentina', datetime.datetime(2020, 1, 7, 0, 0), None, None, None, 45195777.0),
 ('Mexico', datetime.datetime(2020, 1, 7, 0, 0), None, None, None, 128932753.0),
 ('Peru', datetime.datetime(2020, 1, 7, 0, 0), None, None, None, 32971846.0),
 ('Thailand', datetime.datetime(2020, 1, 7, 0, 0), None, None, None, 69799978.0),
 ('Thailand', datetime.datetime(2020, 1, 8, 0, 0), None, None, None, 69799978.0),
 ('Peru', datetime.datetime(2020, 1, 8, 0, 0), None, None, None, 32971846.0),
 ('Argentina', datetime.datetime(2020, 1, 8, 0, 0), None, None, None, 45195777.0),
 ('Mexico', datetime.datetime(2020, 1, 8, 0, 0), None, None, None, 128932753.0),
 ('Mexico', datetime.datetime(2020, 1, 9, 0, 0), None, None, None, 128932753.0),
 ('Peru', datetime.datetime(2020, 1, 9, 0, 0), None, None, None, 32971846.0),
 ('Thailand', datetime.datetime(2020, 1, 9, 0, 0), None, None, None, 69799978.0),
 ('Argentina', datetime.datetime(2020, 1, 9, 0, 0), None, None, None, 45195777.0),
 ('Argentina', datetime.datetime(2020, 1, 10, 0, 0), None, None, None, 45195777.0),
 ('Mexico', datetime.datetime(2020, 1, 10, 0, 0), None, None, None, 128932753.0),
 ('Thailand', datetime.datetime(2020, 1, 10, 0, 0), None, None, None, 69799978.0),
 ('Peru', datetime.datetime(2020, 1, 10, 0, 0), None, None, None, 32971846.0),
 ('Peru', datetime.datetime(2020, 1, 11, 0, 0), None, None, None, 32971846.0),
 ('Thailand', datetime.datetime(2020, 1, 11, 0, 0), None, None, None, 69799978.0),
 ('Argentina', datetime.datetime(2020, 1, 11, 0, 0), None, None, None, 45195777.0),
 ('Mexico', datetime.datetime(2020, 1, 11, 0, 0), None, None, None, 128932753.0),
 ('Mexico', datetime.datetime(2020, 1, 12, 0, 0), None, None, None, 128932753.0),
 ('Thailand', datetime.datetime(2020, 1, 12, 0, 0), None, None, None, 69799978.0),
 ('Peru', datetime.datetime(2020, 1, 12, 0, 0), None, None, None, 32971846.0),
 ('Argentina', datetime.datetime(2020, 1, 12, 0, 0), None, None, None, 45195777.0),
 ('Argentina', datetime.datetime(2020, 1, 13, 0, 0), None, None, None, 45195777.0),
 ('Mexico', datetime.datetime(2020, 1, 13, 0, 0), None, None, None, 128932753.0),
 ('Peru', datetime.datetime(2020, 1, 13, 0, 0), None, None, None, 32971846.0),
 ('Thailand', datetime.datetime(2020, 1, 13, 0, 0), None, None, None, 69799978.0),
 ('Thailand', datetime.datetime(2020, 1, 14, 0, 0), None, None, None, 69799978.0),
 ('Peru', datetime.datetime(2020, 1, 14, 0, 0), None, None, None, 32971846.0),
 ('Argentina', datetime.datetime(2020, 1, 14, 0, 0), None, None, None, 45195777.0),
 ('Mexico', datetime.datetime(2020, 1, 14, 0, 0), None, None, None, 128932753.0),
 ('Mexico', datetime.datetime(2020, 1, 15, 0, 0), None, None, None, 128932753.0)

BY running the query above we can clearly see that the first ever infected cases and deaths were spoted in china on January 22, 2020. Total death counted that day is 17 and total cases were 548. 

Lets find out the overall death ratio compared to total cases from my home country which is Bangladesh. 

In [27]:
%%sql 

-- Total Cases vs Total Deaths

SELECT location, date, total_cases, total_deaths, (total_deaths/total_cases)*100 AS Death_percentage
FROM covid_deaths
WHERE location like '%desh%' and continent is not null
ORDER BY 5 DESC, 2
LIMIT 100;

 * mysql://root:***@localhost:3306/mydatadb
100 rows affected.


[('Bangladesh', datetime.datetime(2020, 3, 25, 0, 0), 39.0, 5.0, 12.82051282051282),
 ('Bangladesh', datetime.datetime(2020, 4, 4, 0, 0), 70.0, 8.0, 11.428571428571429),
 ('Bangladesh', datetime.datetime(2020, 3, 26, 0, 0), 44.0, 5.0, 11.363636363636363),
 ('Bangladesh', datetime.datetime(2020, 4, 1, 0, 0), 54.0, 6.0, 11.11111111111111),
 ('Bangladesh', datetime.datetime(2020, 4, 2, 0, 0), 56.0, 6.0, 10.714285714285714),
 ('Bangladesh', datetime.datetime(2020, 3, 27, 0, 0), 48.0, 5.0, 10.416666666666668),
 ('Bangladesh', datetime.datetime(2020, 3, 28, 0, 0), 48.0, 5.0, 10.416666666666668),
 ('Bangladesh', datetime.datetime(2020, 3, 29, 0, 0), 48.0, 5.0, 10.416666666666668),
 ('Bangladesh', datetime.datetime(2020, 4, 7, 0, 0), 164.0, 17.0, 10.365853658536585),
 ('Bangladesh', datetime.datetime(2020, 3, 24, 0, 0), 39.0, 4.0, 10.256410256410255),
 ('Bangladesh', datetime.datetime(2020, 4, 5, 0, 0), 88.0, 9.0, 10.227272727272728),
 ('Bangladesh', datetime.datetime(2020, 3, 30, 0, 0), 49.0, 5.0, 10.204081632653061),
 ('Bangladesh', datetime.datetime(2020, 4, 3, 0, 0), 61.0, 6.0, 9.836065573770492),
 ('Bangladesh', datetime.datetime(2020, 3, 31, 0, 0), 51.0, 5.0, 9.803921568627452),
 ('Bangladesh', datetime.datetime(2020, 4, 6, 0, 0), 123.0, 12.0, 9.75609756097561),
 ('Bangladesh', datetime.datetime(2020, 4, 8, 0, 0), 218.0, 20.0, 9.174311926605505),
 ('Bangladesh', datetime.datetime(2020, 3, 23, 0, 0), 33.0, 3.0, 9.090909090909092),
 ('Bangladesh', datetime.datetime(2020, 3, 21, 0, 0), 25.0, 2.0, 8.0),
 ('Bangladesh', datetime.datetime(2020, 3, 22, 0, 0), 27.0, 2.0, 7.4074074074074066),
 ('Bangladesh', datetime.datetime(2020, 3, 18, 0, 0), 14.0, 1.0, 7.142857142857142),
 ('Bangladesh', datetime.datetime(2020, 4, 10, 0, 0), 424.0, 27.0, 6.367924528301887),
 ('Bangladesh', datetime.datetime(2020, 4, 9, 0, 0), 330.0, 21.0, 6.363636363636363),
 ('Bangladesh', datetime.datetime(2020, 4, 11, 0, 0), 482.0, 30.0, 6.224066390041494),
 ('Bangladesh', datetime.datetime(2020, 3, 19, 0, 0), 17.0, 1.0, 5.88235294117647),
 ('Bangladesh', datetime.datetime(2020, 4, 12, 0, 0), 621.0, 34.0, 5.475040257648954),
 ('Bangladesh', datetime.datetime(2020, 3, 20, 0, 0), 20.0, 1.0, 5.0),
 ('Bangladesh', datetime.datetime(2020, 4, 13, 0, 0), 803.0, 39.0, 4.85678704856787),
 ('Bangladesh', datetime.datetime(2020, 4, 14, 0, 0), 1012.0, 46.0, 4.545454545454546),
 ('Bangladesh', datetime.datetime(2020, 4, 17, 0, 0), 1838.0, 75.0, 4.080522306855277),
 ('Bangladesh', datetime.datetime(2020, 4, 15, 0, 0), 1231.0, 50.0, 4.061738424045491),
 ('Bangladesh', datetime.datetime(2020, 4, 18, 0, 0), 2144.0, 84.0, 3.9179104477611943),
 ('Bangladesh', datetime.datetime(2020, 4, 16, 0, 0), 1572.0, 60.0, 3.816793893129771),
 ('Bangladesh', datetime.datetime(2020, 4, 19, 0, 0), 2456.0, 91.0, 3.705211726384365),
 ('Bangladesh', datetime.datetime(2020, 4, 20, 0, 0), 2948.0, 101.0, 3.426051560379918),
 ('Bangladesh', datetime.datetime(2020, 4, 21, 0, 0), 3382.0, 110.0, 3.252513305736251),
 ('Bangladesh', datetime.datetime(2020, 4, 22, 0, 0), 3772.0, 120.0, 3.1813361611876987),
 ('Bangladesh', datetime.datetime(2020, 4, 23, 0, 0), 4186.0, 127.0, 3.0339225991399905),
 ('Bangladesh', datetime.datetime(2020, 4, 25, 0, 0), 4998.0, 140.0, 2.801120448179272),
 ('Bangladesh', datetime.datetime(2020, 4, 24, 0, 0), 4689.0, 131.0, 2.793772659415654),
 ('Bangladesh', datetime.datetime(2020, 4, 26, 0, 0), 5416.0, 145.0, 2.6772525849335302),
 ('Bangladesh', datetime.datetime(2020, 4, 27, 0, 0), 5913.0, 152.0, 2.5706071368171823),
 ('Bangladesh', datetime.datetime(2020, 4, 28, 0, 0), 6462.0, 155.0, 2.3986381925100586),
 ('Bangladesh', datetime.datetime(2020, 4, 29, 0, 0), 7103.0, 163.0, 2.2948050119667744),
 ('Bangladesh', datetime.datetime(2020, 4, 30, 0, 0), 7667.0, 168.0, 2.19120907786618),
 ('Bangladesh', datetime.datetime(2020, 5, 1, 0, 0), 8238.0, 170.0, 2.0636076717649914),
 ('Bangladesh', datetime.datetime(2020, 5, 2, 0, 0), 8790.0, 175.0, 1.9908987485779295),
 ('Bangladesh', datetime.da

Now we gonna look at the total infected ratio compared to overall population in Bangladesh.  

In [21]:
%%sql 

SELECT location, date, total_cases, population, (total_cases/population)*100 AS cases_percentage
FROM covid_deaths
WHERE location like '%desh%' and continent is not null
ORDER BY 5 DESC, 2
LIMIT 100;

 * mysql://root:***@localhost:3306/mydatadb
100 rows affected.


[('Bangladesh', datetime.datetime(2021, 4, 30, 0, 0), 759132.0, 164689383.0, 0.46094774670447336),
 ('Bangladesh', datetime.datetime(2021, 4, 29, 0, 0), 756955.0, 164689383.0, 0.4596258642853741),
 ('Bangladesh', datetime.datetime(2021, 4, 28, 0, 0), 754614.0, 164689383.0, 0.4582044004621718),
 ('Bangladesh', datetime.datetime(2021, 4, 27, 0, 0), 751659.0, 164689383.0, 0.45641011357726685),
 ('Bangladesh', datetime.datetime(2021, 4, 26, 0, 0), 748628.0, 164689383.0, 0.45456967921241165),
 ('Bangladesh', datetime.datetime(2021, 4, 25, 0, 0), 745322.0, 164689383.0, 0.45256226383457887),
 ('Bangladesh', datetime.datetime(2021, 4, 24, 0, 0), 742400.0, 164689383.0, 0.45078801467123114),
 ('Bangladesh', datetime.datetime(2021, 4, 23, 0, 0), 739703.0, 164689383.0, 0.4491503863366833),
 ('Bangladesh', datetime.datetime(2021, 4, 22, 0, 0), 736074.0, 164689383.0, 0.4469468441690622),
 ('Bangladesh', datetime.datetime(2021, 4, 21, 0, 0), 732060.0, 164689383.0, 0.4445095285832724),
 ('Bangladesh', datetime.datetime(2021, 4, 20, 0, 0), 727780.0, 164689383.0, 0.441910696817657),
 ('Bangladesh', datetime.datetime(2021, 4, 19, 0, 0), 723221.0, 164689383.0, 0.43914245522432976),
 ('Bangladesh', datetime.datetime(2021, 4, 18, 0, 0), 718950.0, 164689383.0, 0.4365490882918664),
 ('Bangladesh', datetime.datetime(2021, 4, 17, 0, 0), 715252.0, 164689383.0, 0.43430364907008),
 ('Bangladesh', datetime.datetime(2021, 4, 16, 0, 0), 711779.0, 164689383.0, 0.4321948306770935),
 ('Bangladesh', datetime.datetime(2021, 4, 15, 0, 0), 707362.0, 164689383.0, 0.42951281200683106),
 ('Bangladesh', datetime.datetime(2021, 4, 14, 0, 0), 703170.0, 164689383.0, 0.42696741416536854),
 ('Bangladesh', datetime.datetime(2021, 4, 13, 0, 0), 697985.0, 164689383.0, 0.4238190630661358),
 ('Bangladesh', datetime.datetime(2021, 4, 12, 0, 0), 691957.0, 164689383.0, 0.4201588392616663),
 ('Bangladesh', datetime.datetime(2021, 4, 11, 0, 0), 684756.0, 164689383.0, 0.41578636553638676),
 ('Bangladesh', datetime.datetime(2021, 4, 10, 0, 0), 678937.0, 164689383.0, 0.4122530473017802),
 ('Bangladesh', datetime.datetime(2021, 4, 9, 0, 0), 673594.0, 164689383.0, 0.4090087580205458),
 ('Bangladesh', datetime.datetime(2021, 4, 8, 0, 0), 666132.0, 164689383.0, 0.40447780413385853),
 ('Bangladesh', datetime.datetime(2021, 4, 7, 0, 0), 659278.0, 164689383.0, 0.4003160300867725),
 ('Bangladesh', datetime.datetime(2021, 4, 6, 0, 0), 651652.0, 164689383.0, 0.3956854947959821),
 ('Bangladesh', datetime.datetime(2021, 4, 5, 0, 0), 644439.0, 164689383.0, 0.39130573462649987),
 ('Bangladesh', datetime.datetime(2021, 4, 4, 0, 0), 637364.0, 164689383.0, 0.38700976856534824),
 ('Bangladesh', datetime.datetime(2021, 4, 3, 0, 0), 630277.0, 164689383.0, 0.38270651605999395),
 ('Bangladesh', datetime.datetime(2021, 4, 2, 0, 0), 624594.0, 164689383.0, 0.3792557775263509),
 ('Bangladesh', datetime.datetime(2021, 4, 1, 0, 0), 617764.0, 164689383.0, 0.3751085763676703),
 ('Bangladesh', datetime.datetime(2021, 3, 31, 0, 0), 611295.0, 164689383.0, 0.37118057573875296),
 ('Bangladesh', datetime.datetime(2021, 3, 30, 0, 0), 605937.0, 164689383.0, 0.3679271784022653),
 ('Bangladesh', datetime.datetime(2021, 3, 29, 0, 0), 600895.0, 164689383.0, 0.364865657429781),
 ('Bangladesh', datetime.datetime(2021, 3, 28, 0, 0), 595714.0, 164689383.0, 0.36171973514528255),
 ('Bangladesh', datetime.datetime(2021, 3, 27, 0, 0), 591806.0, 164689383.0, 0.3593467831499496),
 ('Bangladesh', datetime.datetime(2021, 3, 26, 0, 0), 588132.0, 164689383.0, 0.35711591681656857),
 ('Bangladesh', datetime.datetime(2021, 3, 25, 0, 0), 584395.0, 164689383.0, 0.3548467966511235),
 ('Bangladesh', datetime.datetime(2021, 3, 24, 0, 0), 580808.0, 164689383.0, 0.35266875703821177),
 ('Bangladesh', datetime.datetime(2021, 3, 23, 0, 0), 577241.0, 164689383.0, 0.35050286149897103),
 ('Bangladesh', datetime.datetime(2021, 3, 22, 0, 0), 573687.0, 164689383.0, 0.3483448596076166),
 ('Bangladesh', datetime.datetime(2021, 3, 21, 0, 0), 570878.0, 164689383.0

In above queries we used the condition "Continenet is not null" in WHERE statetement is because, If we look at the dataset closely we find that the location column contain 'Africa', 'Asia', 'Europe' as a country. Only when the continent is null. To be more specific and acurate we used "continent is not null" in the WHERE statement.

Lets have look!

In [15]:
%%sql 

SELECT *
FROM covid_deaths
WHERE continent is null
LIMIT 100;

 * mysql://root:***@localhost:3306/mydatadb
100 rows affected.


[('OWID_AFR', None, 'Africa', datetime.datetime(2020, 2, 13, 0, 0), None, 0.0, None, None, 0.0, None, None, 0.0, None, None, 0.0, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 1340598113.0, None, None, None, None, None, None, None, None, None, None, None, None, None, None),
 ('OWID_AFR', None, 'Africa', datetime.datetime(2020, 2, 14, 0, 0), 1.0, 1.0, None, None, 0.0, None, 0.001, 0.001, None, None, 0.0, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 1340598113.0, None, None, None, None, None, None, None, None, None, None, None, None, None, None),
 ('OWID_AFR', None, 'Africa', datetime.datetime(2020, 2, 15, 0, 0), 1.0, 0.0, None, None, 0.0, None, 0.001, 0.0, None, None, 0.0, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 1340598113.0, None, None, None, None, None, None, None, None, None, None, None, None, None, None),
 ('OWID_AFR', None, 'Africa', datetime.datetime(2020, 2, 16, 0, 0), 1.0, 0.0, None, None, 0.0, None, 0.001, 0.0, None, None, 0.0, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 1340598113.0, None, None, None, None, None, None, None, None, None, None, None, None, None, None),
 ('OWID_AFR', None, 'Africa', datetime.datetime(2020, 2, 17, 0, 0), 1.0, 0.0, None, None, 0.0, None, 0.001, 0.0, None, None, 0.0, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 1340598113.0, None, None, None, None, None, None, None, None, None, None, None, None, None, None),
 ('OWID_AFR', None, 'Africa', datetime.datetime(2020, 2, 18, 0, 0), 1.0, 0.0, None, None, 0.0, None, 0.001, 0.0, None, None, 0.0, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 1340598113.0, None, None, None, None, None, None, None, None, None, None, None, None, None, None),
 ('OWID_AFR', None, 'Africa', datetime.datetime(2020, 2, 19, 0, 0), 1.0, 0.0, 0.143, None, 0.0, 0.0, 0.001, 0.0, 0.0, None, 0.0, 0.0, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 1340598113.0, None, None, None, None, None, None, None, None, None, None, None, None, None, None),
 ('OWID_AFR', None, 'Africa', datetime.datetime(2020, 2, 20, 0, 0), 1.0, 0.0, 0.143, None, 0.0, 0.0, 0.001, 0.0, 0.0, None, 0.0, 0.0, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 1340598113.0, None, None, None, None, None, None, None, None, None, None, None, None, None, None),
 ('OWID_AFR', None, 'Africa', datetime.datetime(2020, 2, 21, 0, 0), 1.0, 0.0, 0.0, None, 0.0, 0.0, 0.001, 0.0, 0.0, None, 0.0, 0.0, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 1340598113.0, None, None, None, None, None, None, None, None, None, None, None, None, None, None),
 ('OWID_AFR', None, 'Africa', datetime.datetime(2020, 2, 22, 0, 0), 1.0, 0.0, 0.0, None, 0.0, 0.0, 0.001, 0.0, 0.0, None, 0.0, 0.0, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 1340598113.0, None, None, None, None, None, None, None, None, None, None, None,

Now, Lets go global and find those countries has a most highest infection rate compared to their population

In [20]:
%%sql 

SELECT location, population, MAX(total_cases) AS highestCases, MAX((total_cases/population)*100) AS highestInfectedRate
FROM covid_deaths
WHERE continent is not null
GROUP BY location, population
ORDER BY 4 DESC;


 * mysql://root:***@localhost:3306/mydatadb
210 rows affected.


[('Andorra', 77265.0, 13232.0, 17.125477253607713),
 ('Montenegro', 628062.0, 97389.0, 15.506271673815641),
 ('Czechia', 10708982.0, 1630758.0, 15.22794603632726),
 ('San Marino', 33938.0, 5066.0, 14.92722022511639),
 ('Slovenia', 2078932.0, 240292.0, 11.558434811720634),
 ('Luxembourg', 625976.0, 67205.0, 10.736034608355592),
 ('Bahrain', 1701583.0, 176934.0, 10.398199793956568),
 ('Serbia', 6804596.0, 689557.0, 10.133694932072382),
 ('United States', 331002647.0, 32346971.0, 9.772420641699581),
 ('Israel', 8655541.0, 838481.0, 9.687216547180586),
 ('Sweden', 10099270.0, 973604.0, 9.640340341430619),
 ('Estonia', 1326539.0, 122019.0, 9.198297223074482),
 ('Lithuania', 2722291.0, 247269.0, 9.083121532562096),
 ('Netherlands', 17134873.0, 1522973.0, 8.888148747878084),
 ('Belgium', 11589616.0, 990229.0, 8.544105343956176),
 ('Panama', 4314768.0, 364576.0, 8.449492533549892),
 ('France', 68147687.0, 5677835.0, 8.33166208561121),
 ('Portugal', 10196707.0, 836493.0, 8.203560227826493),
 ('Croatia', 4105268.0, 332183.0, 8.091627635516122),
 ('Hungary', 9660350.0, 779348.0, 8.067492378640525),
 ('Georgia', 3989175.0, 310310.0, 7.778801381237975),
 ('Lebanon', 6825442.0, 526578.0, 7.714928937935447),
 ('Liechtenstein', 38137.0, 2926.0, 7.672339198154024),
 ('Switzerland', 8654618.0, 659974.0, 7.625686078807868),
 ('Spain', 46754783.0, 3524077.0, 7.537361471659487),
 ('Cyprus', 875899.0, 65233.0, 7.447548176216665),
 ('Poland', 37846605.0, 2792142.0, 7.377523030137049),
 ('North Macedonia', 2083380.0, 152367.0, 7.313452178671199),
 ('Armenia', 2963234.0, 216064.0, 7.291493010676849),
 ('Qatar', 2881060.0, 205652.0, 7.138067239141149),
 ('Slovakia', 5459643.0, 382211.0, 7.000659200610736),
 ('Jordan', 10203140.0, 711373.0, 6.9720987852759055),
 ('Brazil', 212559409.0, 14659011.0, 6.896430070522072),
 ('Austria', 9006400.0, 618870.0, 6.871446971042813),
 ('Malta', 441539.0, 30292.0, 6.860549124765876),
 ('Italy', 60461828.0, 4022653.0, 6.653211014394074),
 ('Argentina', 45195777.0, 2977363.0, 6.587701766915082),
 ('United Kingdom', 67886004.0, 4432246.0, 6.528954038891434),
 ('Kuwait', 4270563.0, 273991.0, 6.415805129206617),
 ('Latvia', 1886202.0, 118558.0, 6.285540997199663),
 ('Chile', 19116209.0, 1198245.0, 6.2682145816673165),
 ('Monaco', 39244.0, 2454.0, 6.253185200285394),
 ('Moldova', 4033963.0, 250837.0, 6.218128426066377),
 ('Bosnia and Herzegovina', 3280815.0, 198461.0, 6.049137180852928),
 ('Seychelles', 98340.0, 5873.0, 5.972137482204596),
 ('Bulgaria', 6948445.0, 404380.0, 5.819719376061839),
 ('Palestine', 5101416.0, 296462.0, 5.811366883233989),
 ('Turkey', 84339067.0, 4820591.0, 5.715727208601916),
 ('Uruguay', 3473727.0, 198428.0, 5.71225084757668),
 ('Colombia', 50882884.0, 2859724.0, 5.620208162729141),
 ('Maldives', 540542.0, 29835.0, 5.519460097457737),
 ('Romania', 19237682.0, 1055265.0, 5.485406193947899),
 ('Peru', 32971846.0, 1799445.0, 5.457519727588197),
 ('Kosovo', 1932774.0, 103638.0, 5.362137528754009),
 ('United Arab Emirates', 9890400.0, 520236.0, 5.260009706381946),
 ('Ireland', 4937796.0, 248870.0, 5.040102912311484),
 ('Costa Rica', 5094114.0, 250991.0, 4.927078585206377),
 ('Ukraine', 43733759.0, 2124070.0, 4.856820105493333),
 ('Albania', 2877800.0, 131085.0, 4.555042046007367),
 ('Denmark', 5792203.0, 251249.0, 4.33771053949594),
 ('Cape Verde', 555988.0, 23882.0, 4.295416447837003),
 ('Germany', 83783945.0, 3405365.0, 4.064460082417938),
 ('Paraguay', 7132530.0, 279077.0, 3.912735032309713),
 ('Belarus', 9449321.0, 358738.0, 3.7964420935641834),
 ('Oman', 5106622.0, 193253.0, 3.7843607770459613),
 ('Vatican', 809.0, 27.0, 3.3374536464771323),
 ('Greece', 10423056.0, 345033.0, 3.3102863497999047),
 ('Russia', 145934460.0, 4750755.0, 3.2554031446719303),
 ('Canada', 37742157.0, 1228367.0, 3.254628504671845),
 ('Belize', 397621.0, 12668.0, 3.1859484282771784),
 ('Azerbaijan', 10139175.0, 319109.0, 3.1472876244862134),
 ('Iran', 83992953.0, 2499077.0, 2.9753412765473315),
 ('South Africa', 59308690.0,

As we see, Andorra, Montenegro and Czechia has the highest infected cases among 210 countries from world.  

Now lets see which countries had the most death. 

In [23]:
%%sql

Select Location, max(total_deaths) as highestDeathCount
From covid_deaths
Where continent is not null
Group by Location
Order by 2 DESC;

 * mysql://root:***@localhost:3306/mydatadb
210 rows affected.


[('United States', 576232.0),
 ('Brazil', 403781.0),
 ('Mexico', 216907.0),
 ('India', 211853.0),
 ('United Kingdom', 127775.0),
 ('Italy', 120807.0),
 ('Russia', 108290.0),
 ('France', 104675.0),
 ('Germany', 83097.0),
 ('Spain', 78216.0),
 ('Colombia', 73720.0),
 ('Iran', 71758.0),
 ('Poland', 67502.0),
 ('Argentina', 63865.0),
 ('Peru', 61477.0),
 ('South Africa', 54350.0),
 ('Ukraine', 45923.0),
 ('Indonesia', 45521.0),
 ('Turkey', 40131.0),
 ('Czechia', 29267.0),
 ('Romania', 28109.0),
 ('Hungary', 27540.0),
 ('Chile', 26353.0),
 ('Belgium', 24230.0),
 ('Canada', 24220.0),
 ('Ecuador', 18631.0),
 ('Pakistan', 17957.0),
 ('Netherlands', 17398.0),
 ('Philippines', 17234.0),
 ('Portugal', 16974.0),
 ('Bulgaria', 16399.0),
 ('Iraq', 15465.0),
 ('Sweden', 14048.0),
 ('Egypt', 13339.0),
 ('Bolivia', 12951.0),
 ('Slovakia', 11684.0),
 ('Bangladesh', 11450.0),
 ('Tunisia', 10722.0),
 ('Switzerland', 10632.0),
 ('Greece', 10381.0),
 ('Austria', 10205.0),
 ('Japan', 10199.0),
 ('Morocco', 9023.0),
 ('Jordan', 8836.0),
 ('Bosnia and Herzegovina', 8551.0),
 ('Guatemala', 7524.0),
 ('Lebanon', 7278.0),
 ('Croatia', 7081.0),
 ('Saudi Arabia', 6957.0),
 ('Paraguay', 6385.0),
 ('Israel', 6363.0),
 ('Serbia', 6362.0),
 ('Panama', 6232.0),
 ('Moldova', 5812.0),
 ('Honduras', 5281.0),
 ('Ireland', 4903.0),
 ('North Macedonia', 4855.0),
 ('China', 4845.0),
 ('Azerbaijan', 4517.0),
 ('Slovenia', 4250.0),
 ('Georgia', 4110.0),
 ('Armenia', 4108.0),
 ('Lithuania', 3923.0),
 ('Ethiopia', 3688.0),
 ('Dominican Republic', 3480.0),
 ('Kazakhstan', 3334.0),
 ('Nepal', 3279.0),
 ('Algeria', 3253.0),
 ('Palestine', 3249.0),
 ('Costa Rica', 3231.0),
 ('Myanmar', 3209.0),
 ('Libya', 3029.0),
 ('Kenya', 2724.0),
 ('Afghanistan', 2625.0),
 ('Uruguay', 2616.0),
 ('Belarus', 2542.0),
 ('Denmark', 2484.0),
 ('Albania', 2394.0),
 ('Sudan', 2349.0),
 ('Venezuela', 2136.0),
 ('Kosovo', 2134.0),
 ('Latvia', 2131.0),
 ('El Salvador', 2124.0),
 ('Nigeria', 2063.0),
 ('Oman', 2010.0),
 ('South Korea', 1831.0),
 ('Kyrgyzstan', 1604.0),
 ('Syria', 1592.0),
 ('United Arab Emirates', 1587.0),
 ('Zimbabwe', 1567.0),
 ('Kuwait', 1563.0),
 ('Malaysia', 1506.0),
 ('Montenegro', 1495.0),
 ('Zambia', 1251.0),
 ('Yemen', 1226.0),
 ('Estonia', 1161.0),
 ('Malawi', 1148.0),
 ('Cameroon', 1107.0),
 ('Senegal', 1107.0),
 ('Finland', 914.0),
 ('Australia', 910.0),
 ('Mozambique', 814.0),
 ('Luxembourg', 796.0),
 ('Ghana', 779.0),
 ('Jamaica', 778.0),
 ('Democratic Republic of Congo', 766.0),
 ('Norway', 756.0),
 ('Somalia', 713.0),
 ('Botswana', 712.0),
 ('Sri Lanka', 678.0),
 ('Eswatini', 671.0),
 ('Uzbekistan', 649.0),
 ('Bahrain', 646.0),
 ('Cuba', 644.0),
 ('Madagascar', 643.0),
 ('Namibia', 642.0),
 ('Angola', 596.0),
 ('Mali', 484.0),
 ('Qatar', 458.0),
 ('Mauritania', 455.0),
 ('Malta', 413.0),
 ('Uganda', 342.0),
 ('Rwanda', 335.0),
 ('Belize', 323.0),
 ('Lesotho', 316.0),
 ('Cyprus', 308.0),
 ('Guyana', 296.0),
 ("Cote d'Ivoire", 286.0),
 ('Haiti', 254.0),
 ('Cape Verde', 217.0),
 ('Suriname', 204.0),
 ('Thailand', 203.0),
 ('Bahamas', 199.0),
 ('Niger', 191.0),
 ('Nicaragua', 182.0),
 ('Gambia', 174.0),
 ('Chad', 170.0),
 ('Trinidad and Tobago', 169.0),
 ('Burkina Faso', 157.0),
 ('Comoros', 146.0),
 ('Congo', 144.0),
 ('Guinea', 144.0),
 ('Djibouti', 143.0),
 ('Gabon', 139.0),
 ('Andorra', 125.0),
 ('Togo', 123.0),
 ('Mongolia', 115.0),
 ('South Sudan', 115.0),
 ('Equatorial Guinea', 112.0),
 ('Papua New Guinea', 107.0),
 ('Benin', 99.0),
 ('Cambodia', 93.0),
 ('San Marino', 90.0),
 ('Tajikistan', 90.0),
 ('Central African Republic', 88.0),
 ('Liberia', 85.0),
 ('Sierra Leone', 79.0),
 ('Saint Lucia', 74.0),
 ('Maldives', 73.0),
 ('Guinea-Bissau', 67.0),
 ('Liechtenstein', 57.0),
 ('Barbados', 44.0),
 ('Sao Tome and Principe', 35.0),
 ('Vietnam', 35.0),
 ('Antigua and Barbuda', 32.0),
 ('Monaco', 32.0),
 ('Singapore', 30.0),
 ('Iceland', 29.0),
 ('Seychelles', 28.0),
 ('New Zealand', 26.0),
 ('Tanzania', 21.0),
 ('Mauritius', 17.0),
 ('Taiwan', 12.0),
 ('Saint Vincent a